velox/functions/prestosql/aggregates/ApproxDistinctAggregate.cpp

```diff
@@ -111,7 +112,38 @@ struct HllAccumulator {
 
 template <typename T>
 inline uint64_t hashOne(T value) {
-  return XXH64(&value, sizeof(T), 0);
+  if constexpr (std::is_integral_v<T> && !std::is_same_v<T, int64_t>) {
+    int64_t v = value;
+    return XXH64(&v, sizeof(v), 0);
+  } else {
+    return XXH64(&value, sizeof(T), 0);
+  }
+}
+
+template <>
+inline uint64_t hashOne<float>(float value) {
+  uint64_t v;
+  // All NaN values are collapsed to a single "canonical" NaN value. Follows
+  // https://github.com/prestodb/presto/blob/8e34250ac1e0a2f6fb5d08a2842a7d3b5cd95db3/presto-main/src/main/java/com/facebook/presto/type/RealOperators.java#L180-L183
+  if UNLIKELY (std::isnan(value)) {
```

Comment:
Shall we also handle -0.0 here

Reply:
-0.0 has different underlying binary from 0.0, so they're already differentiated. I added a test case for 0.0 and -0.0 for both float and double.

Reply:
They should not be differentiated, and it will be fixed on Presto side as well, so better fix it now while you are on it